File: Matura.java
public class Matura {
  private static int sklej(int i) {
    if(i < 2)
      return 1;
    else
      return i - 1 + (i % 2 == 1 ? sklej(i/2) + sklej((i+1)/2) : 2*sklej(i/2));
  }
  private static int[] sklej_gen(int size) {
    int i;
    int[] tab = new int[size];
    tab[0] = tab[1] = 1;
    for(i = 2; i < size; ++i)
      tab[i] = i - 1 + tab[i/2] + tab[(i+1)/2];
    return tab;
  }
  public static void main(String[] args) {
    int[] tab = sklej_gen(10000);
    for(int i = 0; i < 10000; ++i)
      System.out.println(tab[i]);
  }
}

File: matura.scala
import scala.collection.mutable._

object Matura extends App {
  def sklej(n : Int) : Int = if (n == 1) 1 else n - 1 + sklej(math.floor(n/2.0).toInt) + sklej(math.ceil(n/2.0).toInt)
  def sklejGen(len : Int) : List[Int] = {
    val list = ListBuffer[Int](1, 1)
    for (n <- 2 to len) yield list += n - 1 + list.apply(math.floor(n/2.0).toInt) + list.apply(math.ceil(n/2.0).toInt)
    list.toList
  }

  sklejGen(10000).foreach(Console.println(_)) // Display generated list
  //for (i <- 1 to 100) yield Console.println(sklej(i))
}
